src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -1065,10 +1107,25 @@ private void ImportRawPInvoke(MethodDesc method)
 
         private void ImportCalli(int token)
         {
+            MethodSignature methodSignature = (MethodSignature)_methodIL.GetObject(token);
+            HandleCall(null, methodSignature, ILOpcode.calli, ((ExpressionEntry)_stack.Pop()).ValueAsType(LLVM.PointerType(_universalSignature, 0), _builder));
         }
 
         private void ImportLdFtn(int token, ILOpcode opCode)
         {
+            MethodDesc method = (MethodDesc)_methodIL.GetObject(token);
+
+            if (opCode == ILOpcode.ldvirtftn && method.IsVirtual && method.OwningType.IsInterface)
+            {
+                _dependencies.Add(_compilation.NodeFactory.VirtualMethodUse(method));
```

Comment:
Is there a pattern for when to use `AddVirtualMethodReference` for this?

Reply:
Replaced all of the copy pasta with ```AddVirtualMethodReference```.